ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/submission/FooCommandGenerator.scala

```diff
@@ -64,22 +72,39 @@ final class FooCommandGenerator(
       .filter { case (_, unlikelihood) => randomDraw(unlikelihood) }
       .map(_._1)
 
+  private def pickDivulgeesSorted(): List[Primitive.Party] =
+    divulgeesWithUnlikelihood
+      .filter { case (_, unlikelihood) => randomDraw(unlikelihood) }
+      .map(_._1)
```

Comment:
filter + map => collect

```suggestion
    divulgeesWithUnlikelihood
      .collect {
        case (party, unlikelihood) if randomDraw(unlikelihood) => party
      }

```